test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
why do we return here? Shouldn't we flag this condition as an error?

Instead of `return;` I expect to see something like `FAIL() << "value and derivatives are NaN"`